### PR TITLE
Account testing

### PR DESCRIPTION
**Pull request by povilassta**
Added authentication and account route testing

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+app.use(express.json());
+app.use(express.urlencoded({ extended: false }));
+
+//Routes
+app.use("/accounts", accountsRouter);
+app.use("/categories", categoriesRouter);
+app.use("/", authRouter);
+
+app.use((err, req, res, next) => {
+  if (err.name === "ValidationError") {
```
Comment (Lokki17) [inline app.js:20]: Please, move this middlware (handle error function) in another file.

File: app.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    return res.status(err.statusCode).json({ message: err.message });
+  } else if (err.name === "CastError") {
+    return res.status(400).json({ message: "Provided id is invalid" });
+  } else {
+    res.status(500).json({ message: "Something went wrong" });
+  }
+  next(err);
+});
+
+//app.listen(3000);
```
Comment (Lokki17) [inline app.js:39]: Is app listening some port now?

File: services/transactions.service.js [hunk 3]
```diff
@@ -10,6 +10,7 @@ const transactionsService = {
         return transactions;
       } else {
         const error = new Error("Account not found");
+        error.name = "NotFoundError";
```
Comment (Lokki17) [inline services/transactions.service.js:13]: Maybe it's better to create custom error?

File: tests/auth.test.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
+import app from "../app.js";
+import supertest from "supertest";
+import mongoose from "mongoose";
+import "dotenv/config";
+
+describe("auth", () => {
+  beforeAll(async () => {
+    await mongoose.disconnect();
```
Comment (Lokki17) [inline tests/auth.test.js:8]: Are you using real mongodb connect in tests? It's very bad idea. Use mock

**Review by Lokki17 — CHANGES_REQUESTED** [review]
Fix remarks and merge

File: tests/auth.test.js (hunk 4 above)
Comment (povilassta) [reply tests/auth.test.js:8]: I'm using a separate mongodb purely for testing, with mock data. Is that a bad idea?

Comment (Lokki17) [reply tests/auth.test.js:8]: Maybe I missed something. What instance of mongodb do you use?

Comment (povilassta) [reply tests/auth.test.js:8]: I have two db urls defined in my enviroment variable. One is the real database (bugetifydb) that has real data (not currently obviously), and another one that is called bugetifydb-test, that database has mock data used for testing purposes.

Comment (povilassta) [reply tests/auth.test.js:8]: And in this before all function i connect to it. In 9th line of auth.test.js file.

Comment (Lokki17) [reply tests/auth.test.js:8]: It's better to use in memory db: https://dev.to/paulasantamaria/testing-node-js-mongoose-with-an-in-memory-database-32np

File: errors/conflict.error.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+class ConflictError extends Error {
+  constructor(message) {
```
Comment (Lokki17) [inline errors/conflict.error.js:2]: name = ...
statusCode = 400   

constructor(message) {
    super(message);
  }

File: errors/conflict.error.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+    super(message);
+    this.name = this.constructor.name;
+    this.statusCode = 400;
```
Comment (Lokki17) [inline errors/conflict.error.js:5]: status code for Conflict is 409 (not 400)

File: errors/duplicate.error.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
+class DuplicateError extends Error {
```
Comment (Lokki17) [inline errors/duplicate.error.js:1]: I woudnt use DuplicateError name for 400 status code. It's better to use more common name, f.e. BadRequestError

**Review by Lokki17 — CHANGES_REQUESTED** [review]
Fix remarks and merge

**Review by Lokki17 — APPROVED** [review]

**Pull request merged**